common/minimodel/__init__.py

```diff
@@ -0,0 +1,474 @@
+class Target(ABC):
+    """
+    RA and Dec should be specified in decimal degrees.
+    """
+    name: str
+    ra: float
+    dec: float
+    magnitudes: Set[Magnitude]
+    type: TargetType
+
+    def __post_init__(self):
+        pass
+
+    def guide_speed(self) -> GuideSpeed:
+        """
+        Calculate the guide speed for this target.
+        """
+        pass
+
+
+@dataclass
+class SiderealTarget(Target):
+    """
+    Proper motion must be specified in milliarcseconds / year.
+    Epoch must be the decimal year.
+    """
+    pm_ra: float
+    pm_dec: float
+    epoch: float
+
+    def __post_init__(self):
+        super().__post_init__()
+
+
+@dataclass
+class NonsiderealTarget(Target):
+    des: str
+    tag: TargetTag
+
+    def __post_init__(self):
+        super().__post_init__()
+
+
+@dataclass
+class Resource:
+    id: str
+    name: str
+    description: Optional[str] = None
+
+
+class QAState(Enum):
+    UNKNOWN = auto()
+    USABLE = auto()
+    PASS = auto()
+    FAIL = auto()
+
+
+@dataclass
+class Atom:
+    """
+    The wavelength must be specified in nanometers.
+    """
+    id: str
+    exec_time: timedelta
+    prog_time: timedelta
+    part_time: timedelta
+    observed: bool
+    qa_state: QAState
+    guide_state: bool
+    required_resources: Set[Resource]
+    wavelength: float
+
+
+class ObservationStatus(Enum):
+    NEW = auto()
+    INCLUDED = auto()
+    PROPOSED = auto()
+    APPROVED = auto()
+    FOR_REVIEW = auto()
+    READY = auto()
+    ONGOING = auto()
+    OBSERVED = auto()
+
+
+class Priority(IntEnum):
+    LOW = auto()
+    MEDIUM = auto()
+    HIGH = auto()
+
+
+class TooType(IntEnum):
+    INTERRUPT = auto()
+    RAPID = auto()
+    STANDARD = auto()
+
+
+class SetupTimeType(Enum):
+    FULL = auto()
+    REACQ = auto()
+    NONE = auto()
+
+
+class ObservationClass(Enum):
+    SCIENCE = auto()
+    PROG_CAL = auto()
+    PARTNER_CAL = auto()
+    ACQ = auto()
+    ACQ_CAL = auto()
+
+
+@dataclass
+class Observation:
+    id: str
+    internal_id: str
+    title: str
+    site: Site
+    status: ObservationStatus
+    active: bool
+    priority: Priority
+    setuptime_type: SetupTimeType
+    acq_overhead: timedelta
+    obs_class: ObservationClass
+    exec_time: timedelta
+    program_used: timedelta
+    partner_used: timedelta
+    constraints: Constraints
+    targets: List[Target]
+    guide_stars: Mapping[Resource, Target]
+    sequence: List[Atom]
+    too_type: Optional[TooType] = None
+
+    def total_used(self) -> timedelta:
+        return self.program_used + self.partner_used
+
+    def required_resources(self) -> Set[Resource]:
+        return {r for a in self.sequence for r in a.required_resources}
+
+    def wavelengths(self) -> Set[float]:
+        return {c.wavelength for c in self.sequence}
+
+    def constraints(self) -> Set[Constraints]:
+        return {self.constraints}
+
+
+# Since Python doesn't allow classes to self-reference, we have to make a basic group
```

Comment:
I understand this is an issue with using type annotations??? Using literal string might solve that.